server/src/main/java/org/opensearch/action/search/AbstractSearchAsyncAction.java

```diff
@@ -455,15 +455,17 @@ private void onShardFailure(final int shardIndex, @Nullable SearchShardTarget sh
             .findNext(shardIt, clusterState, e, () -> totalOps.incrementAndGet());
 
         final boolean lastShard = nextShard == null;
-        logger.debug(
-            () -> new ParameterizedMessage(
-                "{}: Failed to execute [{}] lastShard [{}]",
-                shard != null ? shard : shardIt.shardId(),
-                request,
-                lastShard
-            ),
-            e
-        );
+        if (logger.isTraceEnabled()) {
+            logger.trace(
```

Comment:
which shard failed and why might still be helpful in debugging, do you think it would be better to still keep it as debug but log the exception trace in trace mode only.

Reply:
Yeah good point. Even I thought of removing exception and keep it as debug, though we will need to confirm the impact without exception. @jainankitk thoughts?

Reply:
+1 to log shard failure without exception in debug mode. That should ensure the impact from logging is minimal

Reply:
@shwetathareja I have made the change.

Reply:
If trace is enabled you end up logging twice

Reply:
@Bukhtawar It should log only once, right? As if trace is enabled, it will never go to else condition(debug log). So at max we log only once.